### PR TITLE
Fix/toggle event buttons should work only for their layer

### PR DESCRIPTION
**Pull request by mudhoney**
# Summary

- Maintain all event layer visibility states separately, 
- All buttons of event layers are working with their state and preserved in localStorage state.
- Remove all global state of visibility and remove all related codes
- 'd' button is still usable
- remembers all labels state and works correctly with the output minimal
- correctly works with movie and screenshot makers, handle them as before

# Testing

Enter the browsers and the output modes (minimal, normal, embed) this change was tested

| *browser* | *normal* | *minimal* | *embed* | 
| ------- | -------- | -------- | -------- |
| firefox | untested | untested | untested |
| chrome  | tested | tested | untested |
| safari  | untested | untested | untested |
| edge    | untested | untested | untested |

### CODE REVIEW

File: resources/js/Events/EventManager.js [hunk 1]
```diff
@@ -13,18 +14,27 @@ bitwise: true, regexp: true, strict: true, newcap: true, immed: true, maxlen: 12
...
+     * @description Class to manage event queries and data storage.<br><br>
      *
      * Creates a class which queries the HEK API for event data as the
      * application date and time step changes.  This data is stored in
      * the EventType, EventFeatureRecognitionMethod, and Event classes.
      * Queries are optimized to minimize first the number of queries and
      * second the time window/filesize.<br><br>
      *
-     * @constructs
+     * @param {array} eventGlossaryyy
```
Comment (dgarciabriseno) [inline resources/js/Events/EventManager.js]: yyy.

Thank you for adding parameter docs

File: resources/js/Events/EventManager.js [hunk 2]
```diff
@@ -446,11 +446,12 @@ var EventManager = Class.extend({
 
                         self._eventTypes[eventTypeName]._eventFRMs[frmName].domNode.show();
     
-                        // For each event markers of frm , we need to make them visible. 
-                        // because we are hiding them occasionaly
-                        self._eventMarkers.filter(em => {
-                            return em.belongsToFrm(frmName) && em.belongsToEventType(eventTypeName)
-                        }).forEach(m => m.setVisibility(true));
+                        // For each event markers of frm , we are making them visible if they are not hided per user preferences via state
```
Comment (dgarciabriseno) [inline resources/js/Events/EventManager.js]:
```suggestion
                        // For each event markers of frm, we are making them visible if they are not hidden per user preferences via state
```

File: resources/js/Events/EventManager.js [hunk 3]
```diff
@@ -446,11 +446,12 @@ var EventManager = Class.extend({
```
Comment (dgarciabriseno) [inline resources/js/Events/EventManager.js:453]: Can you add a comment above `_toggleEvents` to explain when this function gets called?

File: resources/js/Events/EventManager.js [hunk 4]
```diff
@@ -565,36 +564,55 @@ var EventManager = Class.extend({
         }
     },
 
-    toggleEventLabels: function (event, labelsBtn) {
+    /**
+     * @description validates if this event manager is maintaining a particular , event source like CCMC and HEK
+     * @param {string} name . event source name like CCMC, HEK
+     * @returns boolean
```
Comment (dgarciabriseno) [inline resources/js/Events/EventManager.js]:
```suggestion
     * @returns {boolean}
```

File: resources/js/Events/EventMarker.js [hunk 5]
```diff
@@ -20,15 +21,20 @@ var EventMarker = Class.extend(
     /**
      * @constructs
      * @description Creates an EventMarker
-     * @param {Object} eventLayer EventLayer associated with the EventMarker
+     * @param {JSON} eventGlossary 
+     * @param {string} parentFRM
      * @param {JSON} event Event details
+     * @param {integer} zIndex, zIndex as you know , visibility hierarchy of this marker in html
+     * @param {boolean} labelVisible, set if the labels of this marker is hided
```
Comment (dgarciabriseno) [inline resources/js/Events/EventMarker.js]:
```suggestion
     * @param {integer} zIndex, zIndex as you know, visibility hierarchy of this marker in html
     * @param {boolean} labelVisible, set if the labels of this marker is hidden
```

File: resources/js/Events/EventTree.js [hunk 6]
```diff
@@ -12,13 +13,23 @@ bitwise: true, regexp: true, strict: true, newcap: true, immed: true, maxlen: 12
...
 
-    init: function (id, data, container, eventManager) {
+    /**
+     * @constructs
+     * @description Creates an EventTree
+     * @param {string} id, ID of the tree
+     * @param {JSON} data, all the data to build tree 
+     * @param {dom} container , dom piece to append event tree dom
+     * @param {EventManager} eventManager, event manager this tree is managed by
+     * @param {integer} zIndex, zIndex as you know , visibility hierarchy of this marker in html
```
Comment (dgarciabriseno) [inline resources/js/Events/EventTree.js]:
```suggestion
     * @param {dom} container, dom piece to append event tree dom
     * @param {EventManager} eventManager, event manager this tree is managed by
     * @param {integer} zIndex, zIndex as you know, visibility hierarchy of this marker in html
```

File: resources/js/Events/EventTree.js [hunk 7]
```diff
@@ -328,4 +329,16 @@ var EventTree = Class.extend({
 
         $(document).trigger("change-feature-events-state");
     },
+
+    /**
+     * @description toggles the visibility of frm nodes is tree , if they are not have any child nides
```
Comment (dgarciabriseno) [inline resources/js/Events/EventTree.js]:
```suggestion
     * @description toggles the visibility of frm nodes is tree, if they do not have any child nodes
```

File: resources/js/Events/HelioviewerEventLayerManager.js [hunk 8]
```diff
@@ -27,28 +28,35 @@ var HelioviewerEventLayerManager = EventLayerManager.extend(
...
-
         // Add the event layer
+        let hekTreeConf = Helioviewer.userSettings.getHelioViewerEventLayerSettings('HEK');
+
         this.addEventLayer(
             new HelioviewerEventLayer(this._eventLayers.length, this._requestDate, this.viewportScale,
-                'HEK', true, true)
+                'HEK', hekTreeConf['markers_visible'], hekTreeConf['labels_visible'], hekTreeConf['layer_available_visible'],{"action": "events", "sources": "HEK", "ar_filter": true})
         );
     },
```
Comment (dgarciabriseno) [inline resources/js/Events/HelioviewerEventLayerManager.js]: Does this function get used anywhere? Can it be removed?

This looks like it was supposed to be `_loadStartingLayers`, maybe dead code.

**Review by dgarciabriseno — CHANGES_REQUESTED** [review]

File: resources/js/Events/HelioviewerEventLayerManager.js (hunk 8 above)
Comment (mudhoney) [reply resources/js/Events/HelioviewerEventLayerManager.js]: Absolutely! I tracked it down and removed it with some dead events as well , 
https://github.com/Helioviewer-Project/helioviewer.org/pull/552/commits/049eab58d42356b43f91fbc5acd69e3e92847dc6

File: resources/js/Events/EventLayerManager.js [hunk 9]
```diff
@@ -31,18 +31,6 @@ var EventLayerManager = EventManager.extend(
...
-                   .bind("save-event-layers",            $.proxy(this.save, this))
-                   .bind("add-new-event-layer",          $.proxy(this.addNewLayer, this))
-                   .bind("remove-event-layer",           $.proxy(this._onLayerRemove, this));
-    },
-
-    /**
-     * Remove a specified layer
-     */
-    _onLayerRemove: function (event, id) {
-        this.removeLayer(id);
```
Comment (dgarciabriseno) [inline resources/js/Events/EventLayerManager.js:45]: I love deleting code

**Review by dgarciabriseno — APPROVED** [review]

File: resources/js/Events/EventLayerManager.js (hunk 9 above)
Comment (mudhoney) [reply resources/js/Events/EventLayerManager.js:45]: me too 👍

**Pull request merged**